### ISSUE TITLE
[Enhancement]: Accessibility issue with javascript injected stars for "your rating"

### ISSUE THREAD

**Issue by webdados** [woothemes/woocommerce#58980]
### Describe the solution you'd like

While testing a website with [Accessibility Checker,](https://wordpress.org/plugins/accessibility-checker/) I came across this error:

<img width="900" alt="Image" src="https://github.com/user-attachments/assets/d315ae8e-54a3-4877-9b90-a30ddf558259" />

These links with stars on it are injected by [single-product.js](https://github.com/woocommerce/woocommerce/blob/9.9.4/plugins/woocommerce/client/legacy/js/frontend/single-product.js#L82) to replace the original select element on [single-product-reviews.php](https://github.com/woocommerce/woocommerce/blob/9.9.4/plugins/woocommerce/templates/single-product-reviews.php#L125)

<img width="260" alt="Image" src="https://github.com/user-attachments/assets/375f4830-b6ad-473a-9d83-7c8e26173a49" />

At first I thought this could be false alarm, as role is set to radio on the links, but i investigated a bit further and there are still accessibility issues, and the link tags should be replaced with proper radio ou button elements. Here's an explanation: https://chatgpt.com/share/6853ed94-7a10-8010-849f-31e7da48c707

**Comment by nerrad** [comment]
This has been reviewed and is an [acceptable accessibility fix](https://github.com/woocommerce/woocommerce/pull/53018) for now.
